services/director-v2/src/simcore_service_director_v2/modules/notifier.py

```diff
@@ -0,0 +1,54 @@
+import contextlib
+
+import socketio
+from fastapi import FastAPI
+from fastapi.encoders import jsonable_encoder
+from models_library.api_schemas_directorv2.notifications import ServiceNoMoreCredits
+from models_library.api_schemas_directorv2.socketio import (
+    SOCKET_IO_SERVICE_NO_MORE_CREDITS_EVENT,
+)
+from models_library.api_schemas_webserver.socketio import SocketIORoomStr
+from models_library.projects_nodes_io import NodeID
+from models_library.users import UserID
+from servicelib.fastapi.app_state import SingletonInAppStateMixin
+
+
+class Notifier(SingletonInAppStateMixin):
+    app_state_name: str = "notifier"
+
+    def __init__(self, sio_manager: socketio.AsyncAioPikaManager):
+        self._sio_manager = sio_manager
+
+    async def notify_shutdown_no_more_credits(
+        self, user_id: UserID, node_id: NodeID
+    ) -> None:
+        await self._sio_manager.emit(
+            SOCKET_IO_SERVICE_NO_MORE_CREDITS_EVENT,
+            data=jsonable_encoder(ServiceNoMoreCredits(node_id=node_id)),
+            room=SocketIORoomStr.from_user_id(user_id),
+        )
+
+
+async def publish_shutdown_no_more_credits(
+    app: FastAPI, *, user_id: UserID, node_id: NodeID
+) -> None:
+    notifier: Notifier = Notifier.get_from_app_state(app)
+    await notifier.notify_shutdown_no_more_credits(user_id=user_id, node_id=node_id)
```

Comment:
will the frontend show something meaningfull, other than just the node ID?

Reply:
shut down the message?? I guess you mean the node, and show an error right?

Reply:
Update:
Yes it will shut down the node/service.
Also a nice message is displayed for each services shut down.